ecs-cli/modules/cli/local/network/setup.go

```diff
@@ -122,6 +131,50 @@ func createLocalNetwork(dockerClient networkCreator) {
 	logrus.Infof("Created network %s with ID %s", EcsLocalNetworkName, resp.ID)
 }
 
+func setupLocalEndpointsImage(dockerClient imagePuller) {
+	if localEndpointsImageExists(dockerClient) {
+		return
+	}
+	pullLocalEndpointsImage(dockerClient)
+}
+
+func localEndpointsImageExists(dockerClient imagePuller) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), docker.TimeoutInS)
+	defer cancel()
+
+	args := filters.NewArgs(filters.Arg("reference", localEndpointsImageName))
+	imgs, err := dockerClient.ImageList(ctx, types.ImageListOptions{
+		Filters: args,
+	})
+	if err != nil {
+		logrus.Fatalf("Failed to list images with filters %v due to %v", args, err)
+	}
+
+	for _, img := range imgs {
+		for _, repotag := range img.RepoTags {
+			if strings.HasPrefix(repotag, localEndpointsImageName) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func pullLocalEndpointsImage(dockerClient imagePuller) {
+	ctx, cancel := context.WithTimeout(context.Background(), docker.TimeoutInS)
+	defer cancel()
+
+	logrus.Infof("Pulling image %s", localEndpointsImageName)
+	rc, err := dockerClient.ImagePull(ctx, localEndpointsImageName, types.ImagePullOptions{})
+	if err != nil {
+		logrus.Fatalf("Failed to pull image with err %v", err)
+	}
+	defer rc.Close()
+
+	ioutil.ReadAll(rc)
```

Comment:
How does this work?
https://golang.org/pkg/io/ioutil/#ReadAll seems to return two values, both of which are ignored here...

Reply:
After our discussion offline, I tested with turning off the wifi during img download (see 3. under Manual Tests). Added an error check here, thanks for pointing it out :)